src/libraries/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.Unix.cs

```diff
@@ -130,13 +133,12 @@ public bool MoveNext()
 
                         if (!isSpecialDirectory && _options.AttributesToSkip != 0)
                         {
-                            if ((_options.AttributesToSkip & FileAttributes.ReadOnly) != 0)
-                            {
-                                // ReadOnly is the only attribute that requires hitting entry.Attributes (which hits the disk)
-                                attributes = entry.Attributes;
-                            }
-
-                            if ((_options.AttributesToSkip & attributes) != 0)
+                            // IsSymbolicLink, IsHidden and IsReadOnly will hit the disk if
+                            // caches had not been initialized yet and we could not soft-retrieve the attributes in Initialize
+                            if ((ShouldSkip(FileAttributes.Directory) && isDirectory) ||
+                                (ShouldSkip(FileAttributes.ReparsePoint) && (isSymlink || entry.IsSymbolicLink)) ||
+                                (ShouldSkip(FileAttributes.Hidden) && (isHidden || entry.IsHidden)) ||
+                                (ShouldSkip(FileAttributes.ReadOnly) && (isReadOnly || entry.IsReadOnly)))
```

Comment:
Should we update docs for the new behavior?

Reply:
Which docs specifically?

The user was already expecting to be able to skip the Hidden or ReadOnly attributes, but we had a bug here, preventing the user from skipping them.

My opinion is that there is no need to update the docs, unless we had this bug documented (which would be strange).